File: Python/01_03_Num_Guess_Game.py
# Beginner Project
# Guess number game
# The user guess a number between a range define by the user

import random


def convert_number(i) :
    try :
        converted_number = int(i)
        return converted_number
    except Exception as err_ :
        err_message_ = err_.args
        return err_message_


try :
    user_range = input('Welcome to the num guessing game, choose the max number in range : ')
    user_input = input(f'choose a number between 1 and {user_range} : ')
    range_int = convert_number(user_range)
    num_guess = convert_number(user_input)
    rand_num = random.randint(1, range_int)
    if num_guess == rand_num :
        print(f'User : {user_input}, Computer : {rand_num} You win !')
    else :
        print(f'User : {user_input}, Computer : {rand_num} You lose, try again !')
except Exception as err :
    err_message = err.args
    print(err_message)

File: CSharp/01_03_Num_Guess_Game/01_03_Num_Guess_Game.cs
// Beginner Project
// Guess number game
// The user guess a number between a range define by the user

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;

namespace CS_Code_Projects
{
    internal class Num_Guess_Game
    {
        private readonly Random random = new Random();

        private const string PlayAgainPrompt = "Do you want to play again? (yes to continue, any other key to exit): ";
        private const string ExitMessage = "The player exited the guessing game.";
        private int GetValidInteger(string prompt)
        {
            while (true)
            {
                Console.WriteLine(prompt);
                string userInput = Console.ReadLine();
                if (int.TryParse(userInput, out int number))
                {
                    return number;
                }
                else
                {
                    Console.WriteLine("Please enter a valid integer.");
                }
            }
        }

        private void New_Game()
        {
            while(true)
            {
                try
                {
                    Console.WriteLine("Welcome to the number guessing game!");

                    int range = GetValidInteger("Please enter a positive integer for the range : ");
                    int userGuess = GetValidInteger("Please enter a positive integer for the answer : ");

                    int computerNumber = random.Next(1, range-1);

                    if (userGuess == computerNumber)
                    {
                        Console.WriteLine($"Congratulations! You guessed correctly. Computer's number: {computerNumber}, Your guess: {userGuess}");
                    }
                    else
                    {
                        Console.WriteLine($"Sorry, you guessed wrong. Computer's number: {computerNumber}, Your guess: {userGuess}");
                    }

                    Console.WriteLine(PlayAgainPrompt);
                    string continueChoice = Console.ReadLine();
                    if (!continueChoice.Equals("yes", StringComparison.OrdinalIgnoreCase))
                    {
                        Console.WriteLine(ExitMessage);
                        break;
                    }
                }
                catch (Exception err)
                {
                    Console.WriteLine(err.Message);
                    Console.ReadLine();
                }
                
            }

        }
        static void Main(string[] args)
        {
            Num_Guess_Game new_guess_game = new Num_Guess_Game();
            new_guess_game.New_Game();
        }
    }
}
